docs/source/guides/data.rst

```diff
@@ -252,9 +252,67 @@ set is not available at the time your model was declared. Simply pass the test s
 
 --------------
 
+********************************************
+Accessing DataLoaders within LightningModule
+********************************************
+
+In the case that you require access to the DataLoader or Dataset objects, DataLoaders for each step can be accessed using the ``Trainer`` object:
+
+.. code-block:: python
+
+    def test_step(self, batch, batch_idx, dataloader_idx):
+        test_dl = self.trainer.test_dataloaders[dataloader_idx]
+        test_dataset = test_dl.dataset
+        test_sampler = test_dl.sampler
+        ...
+        # extract metadata, etc. from the dataset:
+        ...
+
+Since Lightning automatically takes care of iterating over the DataLoader, this is generally unnecessary unless you need to access metadata.
+
+Evaluation dataloaders are iterated over sequentially. If you want to iterate over them in parallel, PyTorch Lightning provides a :class:`~pytorch_lightning.trainer.supporters.CombinedLoader` object which supports collections of dataloaders such as list, tuple, or dictionary. The dataloaders can be accessed using in the same way as the provided structure.:
+
+If you provided a list of dataloaders:
+
+.. code-block:: python
+
+    from pytorch_lightning.trainer.supporters import CombinedLoader
+
+
+    def test_dataloader(self):
+        test_dl1 = ...
+        test_dl2 = ...
+        return CombinedLoader([test_dl1, test_dl2])
```

Comment:
The combined dataloaders are already explained above. Let's cut this short and not go into detail again here. This section should just be about accessing, not creating dataloaders.